contracts/modules/VersionManager.sol

```diff
@@ -208,17 +219,25 @@ contract VersionManager is IVersionManager, IModule, BaseFeature, Owned {
     function upgradeWallet(address _wallet, uint256 _toVersion) external override onlyWhenUnlocked(_wallet) {
         require(
             _isFeatureAuthorisedForWallet(_wallet, msg.sender) || // Upgrade triggered by the RelayerManager (from version 1 to version 2,3,4,...)
-            _isWalletUnderInitialization(_wallet) || // Upgrade triggered by WalletFactory or UpgraderToVersionManager (from version 0 to version 1)
+            _isWalletInitializing(_wallet) || // Upgrade triggered by WalletFactory (from version 0 to version 1)
+            IWallet(_wallet).authorised(msg.sender) || // Upgrade triggered by UpgraderToVersionManager (from version 0 to version 1)
             isOwner(_wallet, msg.sender), // Upgrade triggered directly by the owner (from version 1 to version 2,3,4,...)
             "VM: sender may not upgrade wallet"
         );
-        require(_toVersion > 0 && _toVersion <= lastVersion, "VM: Invalid _toVersion");
         uint256 fromVersion = walletVersions[_wallet];
-        require(fromVersion < _toVersion, "VM: Already on new version");
-        walletVersions[_wallet] = _toVersion;
+        uint256 minVersion_ = minVersion;
+        uint256 toVersion;
+        if(toVersion < minVersion_ && _isWalletInitializing(_wallet)) {
```

Comment:
It should be `_toVersion > minVersion`.

Reply:
I believe it's correct: if the version that the WalletFactory is trying to use is lower than the minimal required version, the version is automatically bumped to that minimal required version.

Reply:
Sorry, I meant `_toVersion < minVersion` :-) You are using `toVersion` which hasn't been initialised at that point.